python/tests/simplify.py

```diff
@@ -218,7 +219,7 @@ def record_edge(self, left, right, parent, child):
         else:
             last = self.edge_buffer[child][-1]
             if last.right == left:
-                last.right = right
+                builtins.object.__setattr__(last, "right", right)
```

Comment:
This isn't a pattern I want to encourage people to use. This seems worse than not having Frozen instances.

Reply:
Yes agreed - I could rejig these instances to not use the ugly way. It only really needs to be used in the custom `__init__` and the metadata decorator.